src/borg/archiver.py

```diff
@@ -433,6 +435,13 @@ def item_is_hardlink_master(item):
             return (partial_extract and stat.S_ISREG(item.mode) and
                     item.get('hardlink_master', True) and 'source' not in item)
 
+        if progress:
+            extracted_size = sum(item.file_size() for item in
+                                 archive.iter_items(filter=lambda item: matcher.match(item.path)))
```

Comment:
github hid this, so i’ll try again:
This code is missing strip_components handling, because borg doesn‘t extract files that are cut away completly by stripping components.
I think we should do all filtering in the filter= callable. See #1418 for another reason